tests/scribeAuth.test.ts

```diff
@@ -2,101 +2,211 @@ import { Auth, Tokens } from '@scribelabsai/auth';
 import { HttpRequest } from '@smithy/protocol-http';
 import * as dotenv from 'dotenv';
 import { describe, expect, it } from 'vitest';
+import { authenticator } from 'otplib';
 
 dotenv.config();
 
 const clientId = process.env['CLIENT_ID']!;
 const username = process.env['USER']!;
+const username2 = process.env['USER2']!;
 const password = process.env['PASSWORD']!;
 const userPoolId = process.env['USER_POOL_ID']!;
 const federatedPoolId = process.env['FEDERATED_POOL_ID']!;
+const otp = process.env['OTPCODE']!;
 const access = new Auth({ clientId, userPoolId });
 const poolAccess = new Auth({
   clientId: clientId,
   userPoolId: userPoolId,
   identityPoolId: federatedPoolId,
 });
 
+const mfaActivatedError = 'challenge instead of tokens';
+
 describe('Get tokens', () => {
   it('Username and password passes', async () => {
     const tokens = await access.getTokens({ username, password });
-    expect(assertTokens(tokens)).toBeTruthy();
+    if ('refreshToken' in tokens && 'idToken' in tokens && 'accessToken' in tokens) {
+      expect(assertTokens(tokens)).toBeTruthy();
+    } else {
+      throw new Error(mfaActivatedError);
+    }
   });
   it('Wrong username fails', async () => {
-    await expect(() => access.getTokens({ username: 'username', password })).rejects.toThrowError();
+    await expect(() => access.getTokens({ username: 'username', password })).rejects.toThrow();
   });
   it('Wrong password fails', async () => {
-    await expect(() => access.getTokens({ username, password: 'password' })).rejects.toThrowError();
+    await expect(() => access.getTokens({ username, password: 'password' })).rejects.toThrow();
   });
 
   it('Empty username fails', async () => {
-    await expect(() => access.getTokens({ username: '', password })).rejects.toThrowError();
+    await expect(() => access.getTokens({ username: '', password })).rejects.toThrow();
   });
   it('Empty password fails', async () => {
-    await expect(() => access.getTokens({ username, password: '' })).rejects.toThrowError();
+    await expect(() => access.getTokens({ username, password: '' })).rejects.toThrow();
   });
   it('Empty username and password fails', async () => {
-    await expect(() => access.getTokens({ username: '', password: '' })).rejects.toThrowError();
+    await expect(() => access.getTokens({ username: '', password: '' })).rejects.toThrow();
   });
   it('RefreshToken passes', async () => {
     const refreshToken = await getRefreshToken();
     const tokens = await access.getTokens({ refreshToken });
-    expect(assertTokens(tokens)).toBeTruthy();
+    if ('refreshToken' in tokens && 'idToken' in tokens && 'accessToken' in tokens) {
+      expect(assertTokens(tokens)).toBeTruthy();
+    } else {
+      throw new Error(mfaActivatedError);
```

Comment:
why are we throwing from inside the test?

Reply:
same comment as above